SPECS/grub2/grub2.spec

```diff
@@ -269,9 +269,31 @@ GRUB_PXE_MODULE_SOURCE=%{buildroot}%{_datadir}/grub2-efi/grubaa64-noprefix.efi
 cp $GRUB_MODULE_SOURCE $EFI_BOOT_DIR/$GRUB_MODULE_NAME
 cp $GRUB_PXE_MODULE_SOURCE $EFI_BOOT_DIR/$GRUB_PXE_MODULE_NAME
 
-%post   -p /sbin/ldconfig
+%post
+/sbin/ldconfig
+# When grub2 is installed, we need to populate the defaults file, assuming that a user
+# may invoke grub tools such as grub2-mkconfig for grub updates, in its current state, running
+# grub2-mkconfig on Mariner will brick Mariner, as there is an assumption that the grub.cfg file
+# is setup and configured only once during bootstrap of the image.
+
+# Only update the file if the file exists and is empty, take last known good CMDLINE configuration
+if [ -f /etc/default/grub ] && [ ! -s /etc/default/grub ]; then
+cat << EOF > /etc/default/grub
+# File generated by postscript of grub2 rpm
+GRUB_TIMEOUT=$(grep -w 'set timeout' /boot/grub2/grub.cfg | awk -F'=' 'NR==1{print $NF}')
+GRUB_DISABLE_SUBMENU=true
+GRUB_TERMINAL="serial console"
+GRUB_SERIAL_COMMAND="serial"
+GRUB_CMDLINE_LINUX="rd.auto=1 $(grep mariner_cmdline /boot/mariner.cfg | cut -d= -f2-) lockdown=integrity sysctl.kernel.unprivileged_bpf_disabled=1 $(grep -o "console=[^[:space:]]*" /proc/cmdline |tr '\n' ' ') $(grep systemd_cmdline /boot/systemd.cfg| cut -d= -f2-)"
```

Comment:
definitely should not generate our config based on /boot/mariner.cfg file in running instances; we should actually move our config from static files into grub config and use grub-mkconfig for real to generate grub config in new images

Reply:
I can see that there are several ways that kernel comandlines get set,
- via installkernel
- kernel install post scripts via /boot/mariner.cfg
- during initial image creation and template expansion 
- and grub env, laid out by the installer for systemd.cfg

As grub tools were not installed into the base image, I was under the impression that the intent was to keep grub management the way it is. And with this PR I intended to support the current grub management but add compatibility for grub tools for those that wanted to use it moving forward.

Did you want to push towards using grub tools by default?